frontend/src/features/settings/preferences/fields/generics/BooleanField.tsx

```diff
@@ -1,40 +1,41 @@
 import React from 'react';
-import { useTranslation } from 'react-i18next';
 
 import { FormControlLabel, Switch } from '@mui/material';
 
-interface AutoSelectoEntitiesProps {
+interface BooleanFieldProps {
+  scope: string;
+  name: string;
+  label: string;
   value: boolean;
   onChange: (target: boolean) => void;
-  pollName: string;
 }
 
-const AutoSelectoEntities = ({
+const BooleanField = ({
+  scope,
+  name,
+  label,
   value,
   onChange,
-  pollName,
-}: AutoSelectoEntitiesProps) => {
-  const { t } = useTranslation();
-
+}: BooleanFieldProps) => {
   return (
     <FormControlLabel
       control={
         <Switch
-          name={`${pollName}_comparison__auto_select_entities`}
+          name={`${scope}_${name}`}
           checked={value}
           onChange={() => onChange(!value)}
```

Comment:
As a precaution, I would read the "checked" state from the event itself. (might avoid race condition?)
```suggestion
          onChange={(event) => onChange(event.target.checked)}
```